latch_cli/nextflow/workflow.py

```diff
@@ -0,0 +1,293 @@
+
+        shutil.copytree(
+            Path("/root"),
+            shared_dir,
+            ignore=lambda src, names: ["latch", ".latch", "nextflow", ".nextflow", "work", "results"],
+            ignore_dangling_symlinks=True,
+            dirs_exist_ok=True,
+        )
+
+        cmd = [
+            "/root/nextflow",
+            "run",
+            str(shared_dir / "{nf_script}"),
+            "-work-dir",
+            str(shared_dir),
+            "-profile",
+            "{execution_profile}",
+            "-c",
+            "latch.config",
+{params_to_flags}
+        ]
+
+        print("Launching Nextflow Runtime")
+        print(' '.join(cmd))
+        print(flush=True)
+
+        env = {{
+            **os.environ,
+            "NXF_HOME": "/root/.nextflow",
+            "K8_STORAGE_CLAIM_NAME": pvc_name,
+        }}
+        subprocess.run(
+            cmd,
+            env=env,
+            check=True,
+            cwd=str(shared_dir),
+        )
+    except subprocess.CalledProcessError:
+        remote = LPath(urljoins("{remote_output_dir}", _get_execution_name(), "nextflow.log"))
+        print()
+        print(f"Uploading .nextflow.log to {{remote.path}}")
+        remote.upload_from(shared_dir / ".nextflow.log")
+        import time
+        time.sleep(10 * 60)
+        raise
+    finally:
+        token = os.environ.get("FLYTE_INTERNAL_EXECUTION_ID")
+        if token is None:
+            raise RuntimeError("failed to get execution token")
+
+        headers = {{"Authorization": f"Latch-Execution-Token {{token}}"}}
+        resp = requests.post(
+            "http://nf-dispatcher-service.flyte.svc.cluster.local/finalize",
+            headers=headers,
+            json={{
+                "pvc_name": pvc_name,
+            }}
+        )
+        if resp.status_code != 200:
+            print("Failed to finalize workflow:", resp.status_code)
+
+
+
+@workflow(metadata._nextflow_metadata)
+def {workflow_func_name}({param_signature_with_defaults}) -> None:
+    pvc_name: str = initialize()
+    nextflow_runtime(pvc_name=pvc_name, {param_args})
+
+"""
+
+
+def _get_flags_for_dataclass(name: str, val: Any) -> List[str]:
+    assert is_dataclass(val)
+
+    flags = []
+    for f in fields(val):
+        flags.extend(get_flag(f"{name}.{f.name}", getattr(val, f.name)))
+
+    return flags
+
+
+def get_flag(name: str, val: Any) -> List[str]:
+    flag = f"--{name}"
+
+    if isinstance(val, bool):
+        return [flag] if val else []
+    elif isinstance(val, LatchFile) or isinstance(val, LatchDir):
+        if val.remote_path is not None:
+            return [flag, val.remote_path]
+
+        return [flag, str(val.path)]
+    elif is_dataclass(val):
+        return _get_flags_for_dataclass(name, val)
+    elif isinstance(val, Enum):
+        return [flag, getattr(val, "value")]
+    else:
+        return [flag, str(val)]
+
+
+def generate_nextflow_config(pkg_root: Path):
+    config_path = Path(pkg_root) / "latch.config"
```

Comment:
how does this get read by nextflow?

Reply:
we pass in `-c latch.config` in the command line